Container Security/kernel-eks-implementation-guide.md

```diff
@@ -0,0 +1,242 @@
+           "registryId": "12345678912345",
+           "repositoryName": "falcon-sensor",
+           "repositoryUri": "123456789123.dkr.ecr.eu-west-1.amazonaws.com/falcon-node-sensor",
+           "createdAt": "2021-02-04T10:30:30+00:00",
+           "imageTagMutability": "MUTABLE",
+           "imageScanningConfiguration": {
+               "scanOnPush": false
+           },
+           "encryptionConfiguration": {
+               "encryptionType": "AES256"
+           }
+       }
+   }
+   ```
+
+ - Note the URI of the newly created repository to the environment variable for further use. Falcon Container Sensor will be available under this URI.
+   ```
+   $ FALCON_NODE_IMAGE_URI=123456789123.dkr.ecr.eu-west-1.amazonaws.com/falcon-node-sensor
+   ```
+
+### Step 3: Build falcon-node-sensor container image
+
+ - Provide OAuth2 Client ID and Client Secret for authentication with CrowdStrike Falcon platform. Establishing and retrieving OAuth2 API credentials can be performed at [falcon-console](https://falcon.crowdstrike.com/support/api-clients-and-keys). These credentials will only be used to download sensor, we recommend you create key pair that has permissions only for
+   ```
+   $ FALCON_CLIENT_ID=1234567890ABCDEFG1234567890ABCDEF
+   $ FALCON_CLIENT_SECRET=1234567890ABCDEFG1234567890ABCDEF
+   ```
+
+ - (optional) Provide name of Falcon Cloud you want to used
+   ```
+   $ FALCON_CLOUD=us-1
+   ```
+
+ - Build falcon-node-sensor container for your particular OS that is running on your cluster nodes. By default EKS clusters use Amazon Linux 2
+   ```
+   $ falcon-node-sensor-build amazonlinux2
+   ```
+   To see various build options see [upstream project](https://github.com/CrowdStrike/Dockerfiles).
+
+### Step 4: Push falcon-node-sensor image to the kube registry
+
+ - Push the image to the kube registry
+   ```
+   $ docker tag falcon-node-sensor:latest $FALCON_NODE_IMAGE_URI:latest
+   ```
+   ```
+   $ docker push $FALCON_NODE_IMAGE_URI:latest
+   Using default tag: latest
+   The push refers to repository [123456789123.dkr.ecr.eu-west-1.amazonaws.com/falcon-node-sensor]
+   913513e672ba: Pushed
+   2c5b2be1a19c: Pushed
+   88cd0d79914e: Pushed
+   c20d459170d8: Pushed
+   db978cae6a05: Pushed
+   aeb3f02e9374: Pushed
+   latest: digest: sha256:132eea2728db09c49fecfae78778f11225f6e9818c71c4f5b2321a0dae4d0c95 size: 1572
+   ```
+
+### Step 5: Deploy the DaemonSet using the helm chart
+
+ - Provide CrowdStrike Falcon Customer ID as environment variable. This CID will be used be helm chart to register your cluster nodes to the CrowdStrike Falcon platform.
+   ```
+   $ CID=1234567890ABCDEFG1234567890ABCDEF-12
+   ```
+
+ - Add the CrowdStrike Falcon Helm repository
+   ```
+   $ helm repo add crowdstrike https://crowdstrike.github.io/falcon-helm
+   ```
+
+ - Install CrowdStrike Falcon Helm Chart. Above command will install the CrowdStrike Falcon Helm Chart with the release name falcon-helm in the falcon-system namespace.
```

Comment:
The test references the above command. What does the command below do? What part is the command, what part is expected output?